### PR TITLE
Fix .sh license pager and test .sh installers in interactive mode

### PR DESCRIPTION
**Pull request by dbast**
### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: examples/miniforge/EULA.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+The License has to be read correctly so that dollar signs don't lead to an unset
+parameter error:
+You have to pay US $8, if you can read this.
```
Comment (chenghlee) [inline examples/miniforge/EULA.txt:28]: Should we replace "Anaconda Inc" with "Example Inc", just so this example text is not confused an actual EULA by an actual legal entity?

File: scripts/run_examples.py [hunk 2]
```diff
@@ -28,18 +28,31 @@
 BLACKLIST = []
 WITH_SPACES = {"extra_files", "noconda", "signing", "scripts"}
 
+# .sh installers to also test in interactiv mode
+# (require all to a have License = have same interactive input steps)
+INTERACTIVE_TESTS = ['miniforge']
+# Test runs with even Python version are done in interactive mode, odd in batch mode
+INTERACTIVE_TESTING = (sys.version_info.minor % 2) == 0
```
[the remark this answers is not in the document]
Comment (dbast) [reply scripts/run_examples.py:35]: yeah... this should be improved in the future... this was the easiest way without increasing the CI time and without fundamentally changing the way how run_examples works.

File: constructor/header.sh [hunk 3]
```diff
@@ -517,7 +517,7 @@ else
     find "$PREFIX/pkgs" -type d -empty -exec rmdir {} \; 2>/dev/null || :
 fi
 
-cat <<EOF
+cat <<'EOF'
```
[the remark this answers is not in the document]
Comment (dbast) [reply constructor/header.sh:520]: There are other places outside header.sh where the substitute behaviour is wanted (scripts in the osx folder).

Another location is https://github.com/conda/constructor/blob/main/constructor/utils.py#L123 ... One can want $ signs in .condarc files (possible according to https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html.  not sure if I want to touch that code as part of this PR.

**Comment by dbast** [comment]
@jaimergp yeah, the substitution should be also fixed.

**Comment by dbast** [comment]
Protected now all header.sh against substitution expansion.

**Pull request merged**